### PR TITLE
KAFKA-2844; Separate keytabs for sasl tests

### PR DESCRIPTION
**Pull request by ijuma**
Use a different keytab for server and client in SASL tests

Also:
- Improve approach used to build the JAAS files programmatically
- Delete stale `kafka_jaas.conf` file
- Move `FourLetterWords` to its own file, add `Zk` prefix and clean-up its usage

### CODE REVIEW

**Comment by ijuma** [comment]
./gradlew test passed locally.

File: core/src/test/scala/unit/kafka/zk/FourLetterWords.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.zk
+
+import java.io.IOException
+import java.net.{SocketTimeoutException, Socket, InetAddress, InetSocketAddress}
+
+object FourLetterWords {
```
Comment (harshach) [inline core/src/test/scala/unit/kafka/zk/FourLetterWords.scala]: can we have better naming here. Not sure whats this file going to do by looking at the name.

**Comment by harshach** [comment]
+1. @ijuma can you upmerge this.

**Comment by ijuma** [comment]
Thanks for the review @harshach, I addressed your comment and rebased on trunk. Tests pass locally.

**Comment by ijuma** [comment]
@harshach Since you previously reviewed this and I addressed your comment, would you be able to take a look again?

**Comment by gwenshap** [comment]
LGTM.

Looks awesome.

**Pull request closed without merging**